ray-operator/controllers/ray/common/service.go

```diff
@@ -151,20 +151,40 @@ func BuildHeadServiceForRayService(rayService rayv1.RayService, rayCluster rayv1
 	return service, nil
 }
 
-// BuildServeServiceForRayService builds the service for head node and worker nodes who have healthy http proxy to serve traffics.
+// BuildServeServiceForRayService builds the serve service for Ray service.
 func BuildServeServiceForRayService(rayService rayv1.RayService, rayCluster rayv1.RayCluster) (*corev1.Service, error) {
+	return BuildServeService(rayService, rayCluster, true)
+}
+
+// BuildServeServiceForRayCluster builds the serve service for Ray cluster.
+func BuildServeServiceForRayCluster(rayCluster rayv1.RayCluster) (*corev1.Service, error) {
+	return BuildServeService(rayv1.RayService{}, rayCluster, false)
+}
+
+// BuildServeService builds the service for head node and worker nodes who have healthy http proxy to serve traffics.
+func BuildServeService(rayService rayv1.RayService, rayCluster rayv1.RayCluster, service bool) (*corev1.Service, error) {
+	name := rayCluster.Name
+	namespace := rayCluster.Namespace
+	if service {
+		name = rayService.Name
+		namespace = rayService.Namespace
+	}
+
 	labels := map[string]string{
-		RayServiceLabelKey:               rayService.Name,
-		RayClusterServingServiceLabelKey: utils.GenerateServeServiceLabel(rayService.Name),
+		RayServiceLabelKey:               name,
```

Comment:
Is this label still needed if we create a service for RayCluster?

Reply:
It is used as selector. No harm in using it